franklin/builder/models.py

```diff
@@ -224,15 +224,21 @@ def build(self):
                 "environment_id": self.id
                 }
 
-            response = make_rest_post_call(url, headers, body)
-            if not status.is_success(response.status_code):
-                logger.error("Negative response from Builder: %s",
-                             response.status_code)
-                self.status = self.FAILED
+            if self.can_build():
```

Comment:
even better, add this to line 208 and then move `if self.current_deploy` inside of the `can_build()` function.